ui/selectors/transactions.js

```diff
@@ -56,15 +59,22 @@ export const getCurrentNetworkTransactions = createDeepEqualSelector(
 export const getUnapprovedTransactions = createDeepEqualSelector(
   (state) => {
     const currentNetworkTransactions = getCurrentNetworkTransactions(state);
+    return filterAndShapeUnapprovedTransactions(currentNetworkTransactions);
+  },
+  (transactions) => transactions,
+);
 
-    return currentNetworkTransactions
-      .filter(
-        (transaction) => transaction.status === TransactionStatus.unapproved,
-      )
-      .reduce((result, transaction) => {
-        result[transaction.id] = transaction;
-        return result;
-      }, {});
+// Unlike `getUnapprovedTransactions` and `getCurrentNetworkTransactions`
+// returns the total number of unapproved transactions on all networks
+export const getAllUnapprovedTransactions = createDeepEqualSelector(
+  (state) => {
+    const { transactions } = state.metamask || [];
+    if (!transactions?.length) {
+      return [];
+    }
```

Comment:
```suggestion
    const { transactions } = state.metamask ?? [];
    if (!transactions.length) {
      return {};
    }
```

Reply:
feel free to ignore the nullish coalesce and optional chaining, but returning the empty object is necessary